### DISCUSSION TITLE
How to compile libcurl for i386 ??

### DISCUSSION THREAD

**Discussion by Zorono** [curl/curl#14781] [Q&A] [answered]
I tried compiling **libcurl** for **i386** with the following steps on **Ubuntu 22.04.4 amd64**
```
git clone ...
cd ...
autoreconf -fi
./configure --with-openssl=/usr/include/openssl CFLAGS=-m32 CC=/usr/bin/gcc
```
but it keeps on complaining about `configure: error: /usr/include/openssl is a bad --with-openssl prefix!` and i tried using **--with-openssl** without providing any path but it results in
```
configure: OpenSSL version does not speak QUIC API
configure: OPT_OPENSSL: yes
configure: OPENSSL_ENABLED: 
configure: error: --with-openssl was given but OpenSSL could not be detected
```
please note that **libssl-dev** is already installed

**Comment by dfandrich** [comment, marked as answer]
Read [INSTALL.md](https://github.com/curl/curl/blob/master/docs/INSTALL.md) for some hints in configuring libcurl. Specifically in your case, it's better to set `PKG_CONFIG_PATH` rather than specifying a path with `--with-openssl`. Also, that latter path is the OpenSSL custom installation path, not the path to the headers. Finally, You've (presumably) installed the x86_64 version of libssl-dev but expect to have an x86 version available in order to compile. You'll need to install an x86 version of OpenSSL, and its dependencies (if any) if you're compiling for x86, whether that's from a pre-built package or compiled yourself.
